ledger/participant-integration-api/src/main/scala/platform/indexer/MeteringAggregator.scala

```diff
@@ -200,5 +203,13 @@ class MeteringAggregator(
       .toVector
 
     meteringStore.insertParticipantMetering(participantMetering)(conn)
+
+    meteringStore.deleteTransactionMetering(
+      lastLedgerMeteringEnd.offset,
```

Comment:
If we crash while executing this method, the transaction metrics that went into the accumulated metrics will stay there forever. I would suggest deleting up `to` `thisLedgerMeteringEnd` without the `from`

Reply:
Under normal operation this would make no difference as there should be no transaction metering before the from.  The reason it is there is to match the offsets given to the select clause.  This way we only delete what we have aggregated.

Reply:
I would agree if the metering aggregation and deletion were transactional. Since they aren't we are opening up to teh possibility that one operation will happen without the other. I am fine with postponing this change to a follow-up PR

Reply:
I hope it is transactional.

Reply:
@mziolekda - the `sql` is run via a `JdbcConnectionProvider` which turns off auto-commit and one commits upon successful completion (see link below).

https://github.com/digital-asset/daml/blob/1f24348e2178fb2dab022ff0cf3d75c5dfa484a6/ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/HikariJdbcConnectionProvider.scala#L74.